### ISSUE TITLE
Cabal upper-bound in custom-setup of polysemy.cabal & polysemy-plugin.cabal.

### ISSUE THREAD

**Issue by jchia** [polysemy-research/polysemy#390]
Why does the github version of polysemy.cabal have no upper bound but the version downloaded from hackage has <3.1? Is the polysemy.cabal difference between github & haddock intended? Is the upper bound necessary?

I was able to "stack test" the github version but the haddock version failed to even "stack init" because of the upper bound. After bumping the upper bound, I was able to "stack init" and "stack test" successfully.

polysemy-plugin.cabal has a similar problem.

**Comment by juhp** [comment]
See also commercialhaskell/stackage#5718

**Comment by fendor** [comment]
I think this `Cabal` constraint causes many users to be unable to use latest `polysemy` and `polysemy-plugin`.
The latter's last minor release `0.2.5.2` seems to have fixed a bug that triggered ghc compiler panics: https://gitlab.haskell.org/ghc/ghc/-/issues/14649#note_342155.
Being unable to upgrade for recent ghc triggers the ghc panic quite often for polysemy users, afaict.

**Comment by googleson78** [comment]
I'm still not sure where the constraint has come from, I'm also trying to get rid of it. I think it's something that one of the tools that prepare a hackage upload does. @TheMatten

**Comment by TheMatten** [comment]
Seems like this commit introduced them: https://github.com/polysemy-research/polysemy/commit/76d5f1953e5c62fff53c05115cb2269655195809
So they may have been added manually after all - I guess it will make sense to just drop them in that case and make sure that CI works.

**Comment by googleson78** [comment]
But that commit doesn't change `Cabal` bounds at all, or am I missing something?

**Comment by tek** [comment]
maybe because of the ghc bounds?

**Comment by timjs** [comment]
> I think this `Cabal` constraint causes many users to be unable to use latest `polysemy` and `polysemy-plugin`.

Yes, for me too. Does anybody know why Cabal is restrained to be < 3.1?

**Comment by fendor** [comment]
Can confirm, build is fixed now, thank you very much!
